File: day4/python/rock_paper_scissor.py
import random
# Rock
rock=("""
    _______
---'   ____)
      (_____)
      (_____)
      (____)
---.__(___)
""")

# Paper
paper=("""
     _______
---'    ____)____
           ______)
          _______)
         _______)
---.__________)
""")

# Scissors
scissors=("""
    _______
---'   ____)____
          ______)
       __________)
      (____)
---.__(___)
""")

print("let's play rock paper scissors with the computer.")
#for player
player=(input("Please choose 0 for rock, 1 for papper and 2 for scissors. "))
if player=="0":
    print(f"you have choosen rock\n {rock}")
elif player=="1":
    print(f"you have choosen paper\n {paper}")
elif player=="2":
    print(f"you have choosen scissors\n {scissors}")
else:
    print("please play the game according to the rule.")


#for computer
computer= str(random.randint(0,2))
if computer=="0":
    print(f"computer have choosen rock \n {rock}")
elif computer=="1":
    print(f"computer have choosen paper \n {paper}")
elif computer=="2":
    print(f"computer have choosen scissors \n {scissors}")

#condition to decide winner
if player==computer:
    print("You both have choosen same thing, which makes it draw.")
elif player=="0" and computer=="1":
    print("Computer have won, as paper beats rock.")
elif player=="0" and computer=="2":
    print("You have won, as rock beats scissors.")
elif player=="1" and computer=="0":
    print("Computer have won, as paper beats rock.")
elif player=="2" and computer=="0":
    print("Computer have won, as rock beats scissors.")
elif player=="1" and computer=="2":
    print("Computer have won, as scissors beats papper.")
elif player=="2" and computer=="1":
    print("You have won, as scissors beats papper.")

File: day4/rock-paper-scissor.js
//3rd party module to read the file from user console.
const readlineSync = require("readline-sync");
// # Rock
const rock = `
    _______
---'   ____)
      (_____)
      (_____)
      (____)
---.__(___)
`;

//Paper
const paper = `
     _______
---'    ____)____
           ______)
          _______)
         ______)
---.__________)
`;

//  Scissors
const scissors = `
    _______
---'   ____)____
          ______)
       __________)
      (____)
---.__(___)
`;
console.log("let's play rock paper scissors with the computer.");

console.log("Please choose 0 for rock, 1 for paper and 2 for scissors.");
let person = Number(readlineSync.question());

//had to multiby by 3 as math.floor round down to the lowest interger.
const computer = Math.floor(Math.random() * 3);

//for person
if (person === 0) {
  console.log(`you have choosen rock\n ${rock}`);
} else if (person === 1) {
  console.log(`you have choosen paper\n ${paper}`);
} else if (person === 2) {
  console.log(`you have choosen scissors\n ${scissors}`);
} else {
  console.log(`please play game with the rules.`);
}
//for computer
if (computer === 0) {
  console.log(`computer has choosen rock\n ${rock}`);
} else if (computer === 1) {
  console.log(`computer has choosen paper\n ${paper}`);
} else if (computer === 2) {
  console.log(`computer has choosen scissors\n ${scissors}`);
}
//logic of the game
if (person === computer) {
  console.log("Thats a very tight game, its a draw!");
} else if (person === 0 && computer == 1) {
  console.log("you lose!");
} else if (person === 0 && computer == 2) {
  console.log("you win!");
} else if (person === 1 && computer == 2) {
  console.log("you lose!");
} else if (person === 2 && computer == 1) {
  console.log("you win!");
} else if (person === 1 && computer == 0) {
  console.log("you win!");
} else if (person === 2 && computer == 0) {
  console.log("you lose!");
} else {
  console.log("you have choosen wrong number, so you lose the game.");
}
